src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -425,16 +423,17 @@ void GetAirPathData(EnergyPlusData &state)
 
     // Loop through the primary air systems and obtain the data for each system
     for (AirSysNum = 1; AirSysNum <= NumPrimaryAirSys; ++AirSysNum) {
+        auto &primaryAirSystems = state.dataAirSystemsData->PrimaryAirSystems(AirSysNum);
         NumOASysControllers = 0;
         NumOASysSimpControllers = 0;
         OASysContListNum = 0;
         PackagedUnit(AirSysNum) = false;
-        PrimaryAirSystems(AirSysNum).OASysExists = false; // init Outside Air system connection data to none
-        PrimaryAirSystems(AirSysNum).isAllOA = false;
-        PrimaryAirSystems(AirSysNum).OASysInletNodeNum = 0;
-        PrimaryAirSystems(AirSysNum).OASysOutletNodeNum = 0;
-        PrimaryAirSystems(AirSysNum).NumOAHeatCoils = 0;
-        PrimaryAirSystems(AirSysNum).NumOACoolCoils = 0;
+        primaryAirSystems.OASysExists = false; // init Outside Air system connection data to none
+        primaryAirSystems.isAllOA = false;
+        primaryAirSystems.OASysInletNodeNum = 0;
+        primaryAirSystems.OASysOutletNodeNum = 0;
+        primaryAirSystems.NumOAHeatCoils = 0;
+        primaryAirSystems.NumOACoolCoils = 0;
```

Comment:
Tried to reference areas where array interpretations exceed 3 (although I did do a few with 3 to minimize code).

Reply:
You can add a shortcut for one array dereference if you want. It won't make the code slower. It just adds a line of code to make the shortcut. You should do it for > 3 for performance reasons, but you can do it for fewer for other reasons. There is no penalty.